bundles/org.openhab.binding.groupepsa/src/main/java/org/openhab/binding/groupepsa/internal/things/GroupePSAHandler.java

```diff
@@ -0,0 +1,487 @@
+import org.openhab.binding.groupepsa.internal.rest.api.dto.Properties;
+import org.openhab.binding.groupepsa.internal.rest.api.dto.Safety;
+import org.openhab.binding.groupepsa.internal.rest.api.dto.Service;
+import org.openhab.binding.groupepsa.internal.rest.api.dto.VehicleStatus;
+import org.openhab.binding.groupepsa.internal.rest.exceptions.GroupePSACommunicationException;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerCallback;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.filosganga.geogson.model.Geometry;
+import com.github.filosganga.geogson.model.positions.SinglePosition;
+
+/**
+ * The {@link GroupePSAHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Arjan Mels - Initial contribution
+ */
+@NonNullByDefault
+public class GroupePSAHandler extends BaseThingHandler {
+    private static final long DEFAULT_POLLING_INTERVAL_M = TimeUnit.MINUTES.toMinutes(1);
+    private static final long DEFAULT_ONLINE_INTERVAL_M = TimeUnit.MINUTES.toMinutes(60);
+
+    private final Logger logger = LoggerFactory.getLogger(GroupePSAHandler.class);
+
+    private @Nullable String id = null;
+    private long lastQueryTimeMs = 0L;
+
+    private @Nullable ScheduledFuture<?> groupepsaPollingJob;
+    private long maxQueryFrequencyNanos = TimeUnit.MINUTES.toNanos(1);
+    private long onlineIntervalM;
+
+    @Override
+    protected @Nullable Bridge getBridge() {
+        return super.getBridge();
+    }
+
+    private void pollStatus() {
+        Bridge bridge = getBridge();
+        if (bridge != null && bridge.getStatus() == ThingStatus.ONLINE) {
+            updateGroupePSAState();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    };
+
+    public GroupePSAHandler(Thing thing) {
+        super(thing);
+    }
```

Comment:
What's the purpose of this?

Reply:
Otherwise I get: Implicit super constructor org.openhab.core.thing.binding.BaseThingHandler() is undefined for default constructor. Must define an explicit constructor